<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TechSpecPage</class>
 <widget class="QWidget" name="TechSpecPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>20</number>
   </property>
   <property name="topMargin">
    <number>20</number>
   </property>
   <property name="rightMargin">
    <number>20</number>
   </property>
   <property name="bottomMargin">
    <number>20</number>
   </property>
   <item>
    <widget class="QLabel" name="headerLabel">
     <property name="text">
      <string>Technical Specification &amp; Architecture</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <widget class="QWidget" name="initialChoicePage">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="instructionLabel">
         <property name="text">
          <string>First, choose how to create the Technical Specification document.</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="proposeStackButton">
           <property name="text">
            <string>Let Klyve Propose a Tech Stack</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pmDefineButton">
           <property name="text">
            <string>I Will Provide Technology Guidelines</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pmDefinePage">
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QLabel" name="pmDefineHeaderLabel">
         <property name="text">
          <string>Define Technology Guidelines</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="pmDefineInstructionLabel">
         <property name="text">
          <string>Provide your key technology choices or guidelines below (e.g., 'Use Python with a Flask backend and a SQLite database'). You can also upload supporting documents. The AI will use your input to generate the full technical specification document.</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="pmGuidelinesTextEdit"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="uploadLabel">
           <property name="text">
            <string>Supporting Documents (Optional):</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="uploadPathLineEdit">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="browseFilesButton">
           <property name="text">
            <string>Browse...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="generateFromGuidelinesButton">
           <property name="text">
            <string>Analyze &amp;&amp; Generate Draft</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="reviewPage">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="reviewHeaderLabel">
         <property name="text">
          <string>Review Technical Specification</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTabWidget" name="reviewTabWidget">
         <widget class="QWidget" name="specDraftTab">
          <attribute name="title">
           <string>Specification Draft</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="QTextEdit" name="techSpecTextEdit"/>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="aiAnalysisTab">
          <attribute name="title">
           <string>AI Analysis &amp; Suggestions</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QTextEdit" name="aiAnalysisTextEdit">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="feedbackTab">
          <attribute name="title">
           <string>Your Feedback &amp; Refinements</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_10">
           <item>
            <widget class="QLabel" name="feedbackLabel">
             <property name="text">
              <string>Provide feedback and clarifications below to have the AI generate a new version of the draft.</string>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPlainTextEdit" name="feedbackTextEdit"/>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
        <spacer name="horizontalSpacer">
        <property name="orientation">
            <enum>Qt::Horizontal</enum>
        </property>
        </spacer>
        </item>
        <item>
        <widget class="QPushButton" name="pauseProjectButton">
        <property name="text">
            <string>Pause Project</string>
        </property>
        </widget>
        </item>
        <item>
        <widget class="QPushButton" name="refineButton">
        <property name="text">
            <string>Submit Feedback &amp;&amp; Refine</string>
        </property>
        </widget>
        </item>
        <item>
        <widget class="QPushButton" name="approveButton">
        <property name="text">
            <string>Approve Technical Specification</string>
        </property>
        </widget>
        </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="processingPage">
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="processingLabel">
         <property name="text">
          <string>Processing... Please wait.</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="osSelectionPage">
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QLabel" name="osLabel">
         <property name="text">
          <string>Please select the target Operating System for the application.</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="osComboBox">
         <item>
          <property name="text">
           <string>Windows</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Linux</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>macOS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Android and iOS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Windows and iOS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Linux and Windows</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="generateProposalButton">
           <property name="text">
            <string>Generate Proposal</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>